index.js

```diff
@@ -121,7 +122,9 @@ module.exports = function (str, opts) {
 	str = url.format(urlObj);
 
 	// remove ending `/`
-	str = str.replace(/\/$/, '');
+	if (opts.removeTrailingSlash || !urlObj.pathname || urlObj.pathname === '/') {
```

Comment:
I don't understand the `!urlObj.pathname`. If there's no pathname, there's nothing to strip, so that check seems moot.